STYLE.md

```diff
@@ -0,0 +1,432 @@
+  -> ByteString  -- ^ Data to send
+  -> IO Int      -- ^ Bytes sent
+```
+
+* For functions, the documentation should give enough information to
+apply the function without looking at its definition.
+
+Record example:
+
+```haskell
+-- | Bla bla bla.
+data Person = Person
+  { age  :: !Int     -- ^ Age
+  , name :: !String  -- ^ First name
+  }
+```
+
+For fields that require longer comments, format them this way:
+
+```haskell
+data Record = Record
+  { -- | This is a very very very long comment that is split over
+    -- multiple lines.
+    field1 :: !Text
+
+    -- | This is a second very very very long comment that is split
+    -- over multiple lines.
+  , field2 :: !Int
+  }
+```
+
+#### End-of-Line Comments
+
+Separate end-of-line comments from the code with 2 spaces. Align
+comments for data type definitions. Some examples:
+
+```haskell
+data Parser =
+  Parser
+    !Int         -- ^ Current position
+    !ByteString  -- ^ Remaining input
+
+foo :: Int -> Int
+foo n = salt * 32 + 9
+  where
+    salt = 453645243  -- Magic hash salt.
+```
+
+#### Links
+
+Use in-line links economically.  You are encouraged to add links for
+API names. It is not necessary to add links for all API names in a
+Haddock comment. We therefore recommend adding a link to an API name
+if:
+
+* The user might actually want to click on it for more information (in
+  your opinion), and
+
+* Only for the first occurrence of each API name in the comment (don't
+  bother repeating a link)
+
+Top-down guideline
+------------------
+
+### LANGUAGE extensions section
+
+Write each `LANGUAGE` pragma on its own line, sort them alphabetically and
+align by max width among them.
+
+```haskell
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE Rank2Types       #-}
+{-# LANGUAGE TemplateHaskell  #-}
+```
+
+### Module name
+
+Use singular when naming modules (e.g. use `Data.Map` and
+`Data.ByteString.Internal` instead of `Data.Maps` and
+`Data.ByteString.Internals`). Sometimes it's acceptable to use plural
+(e. g. `Types`, `Instances`).
+
+### Export Lists
+
+Format export lists as follows:
+
+```haskell
+module Data.Set
+  ( Set
+  , empty
+  , singleton
+  , member
+  ) where
+```
+
+Some clarifications:
+
+1. Use 2 spaces indentation for export list.
+2. You can split export list into sections or just write all as single section.
+3. It is strongly adviced to sort each section alpabetically. However,
+   classes, data types and type aliases should be written before
+   functions.
+
+### Imports
+
+Imports should be grouped in the following order:
+
+0. Implicit import of custom prelude (for example [`cardano-prelude`](https://github.com/input-output-hk/cardano-prelude)) if used.
+1. External modules.
+2. External IOHK modules.
+3. Local library modules.
+4. Local test modules
+
+* Put a blank line between each group of imports.
+
+* The imports in each group should be sorted alphabetically, by module name.
+
+* Always use explicit import lists or `qualified` imports.
```

Comment:
I don't think that applies to preludes, does it?

Reply:
Custom preludes seem to be noted as an exception a few lines below: https://github.com/input-output-hk/cardano-node/pull/164/files/027d248e2d99847f3a109c8ae9794be0d8e83e07#diff-32430379d7ba797ddf827c29a4eac4c1R236

Reply:
Ah, my apologies!